utils/tracing/go/tracing.go

```diff
@@ -0,0 +1,104 @@
+// Package tracing provides a simple utility for including opentelemetry and zipkin tracing
+// instrumentation in vHive and Knative workloads
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/trace/zipkin"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func initTracer(tp *sdktrace.TracerProvider) func() {
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+	return func() {
+		_ = tp.Shutdown(context.Background())
+	}
+}
+
+func newZipkinExporter(url string, logger *log.Logger) *zipkin.Exporter {
+	exporter, err := zipkin.NewRawExporter(
+		url,
+		zipkin.WithLogger(logger),
+		zipkin.WithSDKOptions(sdktrace.WithSampler(sdktrace.AlwaysSample())),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return exporter
+}
+
+// InitBasicTracer initialises a basic opentelemetry tracer using a zipkin exporter. The
+// exporter exports to the provided url and the tracer takes on the given name concatenated
+// with the host name (e.g. Knative pod name).
+func InitBasicTracer(url string, serviceName string) func() {
+	var logger = log.New(os.Stderr, "tracer-log", log.Ldate|log.Ltime|log.Llongfile)
+	hostname, ok := os.LookupEnv("HOSTNAME")
+	if !ok {
+		hostname = ""
+	}
+	exporter := newZipkinExporter(url, logger)
```

Comment:
if this call fails, this function needs to return the error to the caller as is and **not** proceed with the next steps